src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -183,9 +183,15 @@ private void MergeTransactionsOnce()
                     {
                         case PendingOperations.CompletedAll:
                         case PendingOperations.ModifiedsSystemDocuments:
-                            tx.Commit();
-                            tx.Dispose();
-                            NotifyOnThreadPool(pendingOps);
+                            try
+                            {
+                                tx.Commit();
+                                tx.Dispose();
+                            }
+                            finally
```

Comment:
What sets the op.Exception here? Are we sure we won't be sending an ok when we actually errored?